src/entry_table.rs

```diff
@@ -67,25 +68,19 @@ impl<const UID_LENGTH: usize, const KWI_LENGTH: usize> EntryTableValue<UID_LENGT
     /// value and returns this UID.
     ///
     /// - `kwi_uid` : KMAC key used to generate Chain Table UIDs.
-    pub(crate) fn next_chain_table_uid<
-        const CHAIN_TABLE_WITH: usize,
-        const BLOCK_LENGTH: usize,
-        const KMAC_KEY_LENGTH: usize,
-        KmacKey: SymKey<KMAC_KEY_LENGTH>,
-    >(
+    pub(crate) fn next_chain_table_uid<const CHAIN_TABLE_WITH: usize, const BLOCK_LENGTH: usize>(
         &mut self,
-        kwi_uid: &KmacKey,
+        kwi_uid: &Aes256GcmKey,
```

Comment:
This seems wrong. You can't change a KMAC key into an AES key.

Reply:
my bad